contracts/contracts/strategies/NativeStaking/NativeStakingSSVStrategy.sol

```diff
@@ -0,0 +1,312 @@
+    /// @param _asset      Address of weth asset
+    /// @return balance    Total value of (W)ETH
+    function checkBalance(address _asset)
+        external
+        view
+        override
+        returns (uint256 balance)
+    {
+        require(_asset == WETH, "Unsupported asset");
+
+        balance =
+            // add the ETH that has been staked in validators
+            activeDepositedValidators *
+            32 ether +
+            // add the WETH in the strategy from deposits that are still to be staked
+            IERC20(WETH).balanceOf(address(this));
+    }
+
+    function pause() external onlyStrategist {
+        _pause();
+    }
+
+    /// @notice Returns bool indicating whether asset is supported by strategy.
+    /// @param _asset The address of the asset token.
+    function supportsAsset(address _asset) public view override returns (bool) {
+        return _asset == WETH;
+    }
+
+    /// @notice Approves the SSV Network contract to transfer SSV tokens for deposits
+    function safeApproveAllTokens() external override {
+        /// @dev Approves the SSV Network contract to transfer SSV tokens for deposits
+        IERC20(SSV_TOKEN).approve(SSV_NETWORK, type(uint256).max);
+    }
+
+    /**
+     * @notice Only accept ETH from the FeeAccumulator and the WETH contract - required when
+     * unwrapping WETH just before staking it to the validator
+     * @dev don't want to receive donations from anyone else as this will
+     * mess with the accounting of the consensus rewards and validator full withdrawals
+     */
+    receive() external payable {
+        require(
+            msg.sender == FEE_ACCUMULATOR_ADDRESS || msg.sender == WETH,
+            "Eth not from allowed contracts"
+        );
+    }
+
+    /***************************************
+                Internal functions
+    ****************************************/
+
+    function _abstractSetPToken(address _asset, address) internal override {}
+
+    /// @dev Convert accumulated ETH to WETH and send to the Harvester.
+    /// Will revert if the strategy is paused for accounting.
+    function _collectRewardTokens() internal override whenNotPaused {
+        // collect ETH from execution rewards from the fee accumulator
+        uint256 executionRewards = FeeAccumulator(FEE_ACCUMULATOR_ADDRESS)
+            .collect();
+
+        // total ETH rewards to be harvested = execution rewards + consensus rewards
+        uint256 ethRewards = executionRewards + consensusRewards;
+
+        require(
+            address(this).balance >= ethRewards,
+            "Insufficient eth balance"
+        );
+
+        if (ethRewards > 0) {
+            // reset the counter keeping track of beacon chain consensus rewards
+            consensusRewards = 0;
+
+            // Convert ETH rewards to WETH
+            IWETH9(WETH).deposit{ value: ethRewards }();
+
+            emit RewardTokenCollected(harvesterAddress, WETH, ethRewards);
```

Comment:
bump emit of event down and transfer up right under the deposit?

Reply:
done